src/routes/forms/submission/submissionName/router.ts

```diff
@@ -25,9 +27,21 @@ submissionNameApiRoute.get(
           .json({ error: "Form submission does not exist" });
       }
 
-      return response.json(formSubmission);
+      const serviceAccountId = request.serviceAccountId;
+      if (!serviceAccountId) {
+        throw new Error(
```

Comment:
Definitely, I think this will part of a larger refactor I would like to get to that separates out the concerns between, routes, middleware, and controllers.